### PR TITLE
Add sliding window method

### PR DESCRIPTION
**Pull request by kborucinski**
Add sliding window method which returns a new list, composed of n-tuples of consecutive elements.

### CODE REVIEW

File: array/slidingWindow.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    {
+      "language": "javascript",
+      "content": "slidingWindow(3)([1, 2, 3, 4, 5]); // ⇒ [[1, 2, 3], [2, 3, 4], [3, 4, 5]]"
+    },
+    {
+      "language": "javascript",
+      "content": "slidingWindow(1)([1, 2, 3, 4, 5, 6]); // ⇒ [[1], [2], [3], [4], [5], [6]]"
+    }
+  ],
+  "questions": ["TODO: List questions that may this function answers."]
```
Comment (marekrozmus) [inline array/slidingWindow.json]: answers => answer

File: array/slidingWindow.test.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/* eslint-env jest */
+// @ts-ignore ambiguous import
+import slidingWindow from "./slidingWindow.ts";
+
+describe("slidingWindow", () => {
+  it("should returns list, composed of 2-tuples of consecutive elements", () => {
```
Comment (marekrozmus) [inline array/slidingWindow.test.ts]: returns => return

File: array/slidingWindow.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      [2, 3, 4, 5],
+      [3, 4, 5, 6]
+    ]);
+  });
+
+  it("should return a wrapped array if the window size is equal to length of the array given as an argument", () => {
+    expect(slidingWindow(6)([1, 2, 3, 4, 5, 6])).toEqual([[1, 2, 3, 4, 5, 6]]);
+  });
+
+  it("should return array of wrapped values in arrays", () => {
```
Comment (przemyslawzalewski) [inline array/slidingWindow.test.ts]: Please correct the test case with the proper condition - "when sliding window length is 1"

File: array/slidingWindow.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      [1],
+      [2],
+      [3],
+      [4],
+      [5],
+      [6]
+    ]);
+  });
+
+  it("should return empty list if window size is greater than length of the array given as an argument", () => {
```
Comment (przemyslawzalewski) [inline array/slidingWindow.test.ts]:
```suggestion
  it("should return empty list if window size is greater than length of the given array", () => {
```

File: array/slidingWindow.test.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+/* eslint-env jest */
+// @ts-ignore ambiguous import
+import slidingWindow from "./slidingWindow.ts";
+
+describe("slidingWindow", () => {
+  it("should return list, composed of 2-tuples of consecutive elements", () => {
```
Comment (przemyslawzalewski) [inline array/slidingWindow.test.ts]:
```suggestion
  it("should an array of pairs of consecutive elements for sliding window length of 2", () => {
```

Please apply a similar fix for other cases (pairs, triplets, quartets, etc.).

File: array/README.md [hunk 6]
```diff
@@ -428,6 +428,34 @@ Skips the given count of elements from the given array.
 ```
 <!-- prettier-ignore-end -->
 
+# slidingWindow
+
+Returns a new list composed of tuples of the given sliding window length of consecutive elements.
```
Comment (przemyslawzalewski) [inline array/README.md]: Please use `array` wording as there are no `lists` actually but plain arrays here in JS.

File: array/slidingWindow.test.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    expect(slidingWindow(2)([1, 2, 3, 4, 5, 6])).toEqual([
+      [1, 2],
+      [2, 3],
+      [3, 4],
+      [4, 5],
+      [5, 6]
+    ]);
+  });
+
+  it("should return an array composed of 3-tuples of consecutive elements for sliding window length of 3", () => {
```
Comment (przemyslawzalewski) [inline array/slidingWindow.test.ts]:
```suggestion
  it("should return an array composed of triples of consecutive elements for sliding window length of 3", () => {
```

File: array/slidingWindow.test.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    expect(slidingWindow(3)([1, 2, 3, 4, 5, 6])).toEqual([
+      [1, 2, 3],
+      [2, 3, 4],
+      [3, 4, 5],
+      [4, 5, 6]
+    ]);
+  });
+
+  it("should return an array composed of 4-tuples of consecutive elements for sliding window length of 4", () => {
```
Comment (przemyslawzalewski) [inline array/slidingWindow.test.ts]:
```suggestion
  it("should return an array composed of quadruples of consecutive elements for sliding window length of 4", () => {
```

**Pull request merged**